pkg/destroy/ibmcloud/disk.go

```diff
@@ -0,0 +1,132 @@
+package ibmcloud
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func (o *ClusterUninstaller) listDisks() ([]cloudResource, error) {
+	o.Logger.Infof("Listing disks")
+
+	result := []cloudResource{}
+	clusterOwnedTag := o.clusterLabelFilter()
+	options := o.vpcSvc.NewListVolumesOptions()
+
+	for {
+		ctx, cancel := o.contextWithTimeout()
+		defer cancel()
+		options.SetLimit(100)
+		resources, _, err := o.vpcSvc.ListVolumesWithContext(ctx, options)
+		if err != nil {
```

Comment:
in case of error for any page we are returning nil, is this expected i.e no volume deletion in case of error or do we need to return the partial list or need to to re-try

Reply:
Lets discuss with IPI team on this